### PR TITLE
[Feature](Nereids)support binary arithmetic function

### PR DESCRIPTION
**Pull request by sohardforaname**
# Proposed changes

Issue Number: close #xxx

## Problem summary

support binary arithmetic functions like:
```sql
add(op1, op2) -> op1 + op2
subtract(op1, op2) -> op1 - op2
multiply(op1, op2) -> op1 * op2
divide(op1, op2) -> op1 / op2
mod(op1, op2) -> op1 % op2
```

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run p0

File: gen/DorisLexer.interp
Comment (morrySnow) [inline gen/DorisLexer.interp:1]: remove these files

**Comment by sohardforaname** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FunctionBinder.java [hunk 1]
```diff
@@ -96,6 +96,12 @@ public Expression visitUnboundFunction(UnboundFunction unboundFunction, Cascades
                 .build()
                 : (List) unboundFunction.getArguments();
 
+        ArithmeticFunctionBinder functionBinder = new ArithmeticFunctionBinder();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FunctionBinder.java:99]: add comment to explain why these founction cannot be treat as normal scalar function

File: regression-test/suites/nereids_p0/sql_functions/test_arith_functions.groovy [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        sql 'select int_divide(1, 1), bitand(1, 1), bitor(2, 2), bitxor(3.0, 2.0), bitnot(3.0)'
+        result([[1L, 1, 2, 1L, -4L]])
+    }
+    test {
+        sql 'select add(k1, k2) + subtract(k2, k3) + multiply(k3, k4), cast(divide(k4, k3) + mod(k4, k3) as bigint) from test order by k1 limit 1'
+        result([[11022916880, 11902L]])
+    }
+//    test {
+//        sql 'select int_divide(k1, k2), bitand(k2, k3), bitor(k3, k4), bitxor(k4, k3), bitnot(k4) from test order by k1'
+//    }
```
[the remark this answers is not in the document]
Comment (sohardforaname) [reply regression-test/suites/nereids_p0/sql_functions/test_arith_functions.groovy:37]: int_divide(k1, k2) cause be core, int_divide(literal, literal) is ok

**Comment by sohardforaname** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ArithmeticFunctionBinder.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * bind arithmetic function
+ */
+public class ArithmeticFunctionBinder {
+    private static final NullLiteral dummyExpression = new NullLiteral();
+    private static final Map<String, Expression> functionToExpression = ImmutableMap.<String, Expression>builder()
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ArithmeticFunctionBinder.java:43]: u should use case insensitive map, or turn to lowercase before call containsKey and get

Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ArithmeticFunctionBinder.java:43]:
```suggestion
    private static final NullLiteral DUMMPY_EXPRESSION = new NullLiteral();
    private static final Map<String, Expression> ARITHMETIC_FUNCTIONS = ImmutableMap.<String, Expression>builder()
```

**Review by morrySnow — CHANGES_REQUESTED** [review]

**Comment by sohardforaname** [comment]
run buildall

**Review by englefly — DISMISSED** [review]

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**